### ISSUE TITLE
Pod creation fails in a cluster with flannel network provider (Arm64)

### ISSUE THREAD

**Issue by LyingChild** [rancher/rancher#25375]
Rancher versions: 2.3.3

Infrastructure Stack versions:
network-services: flannel
scheduler:
kubernetes: v1.16.6

Docker version: (docker version,docker info preferred)

Client:
 Version:      18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:21:06 2018
 OS/Arch:           linux/arm64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:24:22 2018
  OS/Arch:          linux/arm64
  Experimental:     false

docker info
Containers: 70
 Running: 14
 Paused: 0
 Stopped: 56
Images: 10
Server Version: 18.06.1-ce
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.14.0-115.el7a.0.1.aarch64
Operating System: CentOS Linux 7 (AltArch)
OSType: linux
Architecture: aarch64
CPUs: 96
Total Memory: 510.6GiB
Name: centos-node3
ID: GZ3B:W4EE:E25I:HCEM:MXI6:XXHS:AYS5:UI7P:DHHK:GVC2:IABP:S3AG
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

Operating system and kernel: (cat /etc/os-release, uname -r preferred)

NAME="CentOS Linux"
VERSION="7 (AltArch)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (AltArch)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

uname -r
4.14.0-115.el7a.0.1.aarch64

Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO),
Bare-metal

Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)

    Rancher 2.3.3 single node rancher

Environment Template: (Cattle/Kubernetes/Swarm/Mesos)
Kubernetes

Steps to Reproduce:
    Set up according to the  document（https://rancher.com/docs/rancher/v2.x/en/installation/options/arm64-platform/）
    Add Cluster by using "From existing nodes(Custom)"

Results:

    pod coredns is Unavailable. After retry several  times, it changes to Inactive.

**Comment by LyingChild** [comment]
The version of kube-flannel is 0.11.0 when use k8s 1.16.6，this version seems not compatible with arm64. If change the version of k8s to 1.14.9 which use kube-flannel:v0.10.0(compatible with arm64), the above phenomenon no longer occurs, but the coredns-coredns still not working. So that the worker can not join the cluster.
According to hub.docker.com, coredns-coredns does not explicitly support arm64, so does it support arm64?

**Comment by Confusingboat** [comment]
I'm seeing the same here on . It looks like Rancher is just not building the `arm64` image though, as the repo from upstream (`quay.io/coreos/flannel`) has the `arm64` image.

**Comment by LyingChild** [comment]
> I'm seeing the same here on `1.15.10-rancher1-1`. It looks like Rancher is just not building the `arm64` image for flannel `0.11.0` though, as the repo from upstream (`quay.io/coreos/flannel`) has the `arm64` image.

Thanks for reply，I knew that flannel `0.11.0` doesn't support arm64, so that I build cluster with kubernetes v1.14 which uses flannel `0.10.0` and support arm64. 

In fact, rancher/coredns-coredns also does not support arm64，but can update to coredns/coredns:1.6.5 to solve the problem.

Anyway, thanks~

**Comment by LyingChild** [comment]
I have tried add a X86 node (only worker role) to the cluster, it works.
And it failed to use the same command to add a ARM64 node (only work role) .
But there is a bad way to add a Arm64 node that add with control and worker role.  : )

**Comment by pvizeli** [comment]
I run  v1.16.6 with arm64 and downgraded the flannel version and modify the DNS container. Everything works fine. I'm not sure why but flannel and Calico have official full arm64 support (I would like to see Calico because that is the best network layer).

**Comment by BarthV** [comment]
For those interested in a workaround :
* I've rebuilt flannel `v0.11.0` for using Makefile embedded in the github project.
* Created a multiarch manifest for amd64/arm/arm64 architectures (multiarch was missing for both flannel & rancher images)
* Pushed everything to dockerhub `barthv/coreos-flannel-multiarch:v0.11.0`

If ever you need to use it just change system_images in your `cluster.yml` :
```
system_images:
  flannel: barthv/coreos-flannel-multiarch:v0.11.0
```
"It-works-for-me"®

**Comment by alex-s-team** [comment]
can confirm, it works. i had to use rancher/coredns-coredns:1.6.9 (instead of 1.6.5) and barthv/coreos-flannel-multiarch:v0.11.0 to make it fly. it would be great to be able to use the stock images with arm architecture as well.

**Comment by jgato** [comment]
Has this been solved without the workaround? Do you know from which kubernetes version this is solved?

**Comment by BarthV** [comment]
so far I'm not aware of any change on this

**Comment by jgato** [comment]
ok, changing to barthv/coreos-flannel-multiarch:v0.11.0 worked about having flannel.

But still failing with coredns and coredns-autoscaler. I changed to rancher/coredns-coredns:1.6.9  from version 1.3.5 (that was the default installing). But I have this message:

```
Failed to inspect image "rancher/coredns-coredns:1.6.5": rpc error: code = Unknown desc = Error response from daemon: readlink /var/lib/docker/overlay2: invalid argument

``` 
anyone can help me? @alex-s-team did you changed anything else?

**Comment by alex-s-team** [comment]
@jgato : no, i didn't even change the flannel image any more (rancher version 2.4.5, kubernetes version 1.18.3). at the moment i have the following images running:
rancher/coreos-flannel:v0.12.0
rancher/coredns-coredns:1.6.9
note:  i did not change the cluster.yml, i edited the config map for the pod in the system namespace. this gets overwritten every time the rke deploy job runs, but its just two lines of text, really.

**Comment by jgato** [comment]
ummm. I have rancher server 2.4.3 with Kubernetes 1.17.5 as the last version. Would it be possible? only for one minor release jajaja

I have managed to change flannel to coreos-flannel: coreos-flannel-multiarch:v0.11.0

would this be incompabitle with coredns-coredns:1.6.X? Which actually have arm64 support...

with other versions are you using? coredns-autoscaler?

**Comment by alex-s-team** [comment]
rancher/cluster-proportional-autoscaler:1.7.1
rancher/hyperkube:v1.18.3-rancher2

at the moment, all works fine.

**Comment by jgato** [comment]
it seems pretty different, I dont think it would work if I am using 1.17.5 deployment and changing this things :(

**Comment by jgato** [comment]
Worked perfectly with kubernetes rancher 1.18. I have created an hybrid
infrastructure with Intel an arm.

**Comment by BarthV** [comment]
flannel v0.13.0 will be released (soon) with multi-arch images :)

**Comment by BarthV** [comment]
arm64 (and other) builds are now officially available :
https://github.com/coreos/flannel/releases/tag/v0.13.0

**Comment by LyingChild** [comment]
got it, have solved the problem by using flannel v0.13.0, Thanks everyone!
